lib/pure/httpclient.nim

```diff
@@ -930,26 +923,26 @@ proc requestAux(client: HttpClient | AsyncHttpClient, url: string,
       await client.parseBodyFut
       client.parseBodyFut = nil
 
-  await newConnection(client, requestUrl)
+  await newConnection(client, url)
 
   let effectiveHeaders = client.headers.override(headers)
 
   if not effectiveHeaders.hasKey("user-agent") and client.userAgent != "":
     effectiveHeaders["User-Agent"] = client.userAgent
 
-  var headersString = generateHeaders(requestUrl, httpMethod,
+  var headersString = generateHeaders(url, httpMethod,
                                       effectiveHeaders, body, client.proxy)
 
   await client.socket.send(headersString)
   if body != "":
     await client.socket.send(body)
 
-  let getBody = httpMethod.toLowerAscii() notin ["head", "connect"] and
+  let getBody = httpMethod notin ["HEAD", "CONNECT"] and
                 client.getBody
   result = await parseResponse(client, getBody)
 
-proc request*(client: HttpClient | AsyncHttpClient, url: string,
-              httpMethod: string, body = "",
+proc request*(client: HttpClient | AsyncHttpClient, url: Uri | string,
+              httpMethod: HttpMethod | string = "GET", body = "",
```

Comment:
This should probably be HttpGet by default. Also, AFAIK @Araq dislikes this pattern so might be a good idea to ask him for a review.